include/odp/api/spec/schedule.h

```diff
@@ -45,6 +46,95 @@ extern "C" {
  * Maximum schedule group name length in chars including null char
  */
 
+/**
+ * Schedule configuration options
+ */
+typedef struct odp_schedule_config_t {
+
+	/* Number of flows per queue to be supported
+	 * Only valid when scheduler is configured as flow_aware mode
+	 *
+	 * Flows are lightweight entities and packets can be assigned to specific
+	 * flows by the application using odp_event_flow_hash_set() before
+	 * enqueuing the packet into the scheduler.
+	 * This value is valid only when scheduler is configured in flow aware
+	 * mode.
+	 * Depening on the implementation this number might be rounded-off to
+	 * nearest supported value (e.g power of 2)
+	 * This number should be less than maximum flow supported by the
+	 * implementation.
+	 * @see odp_schedule_capability_t
+	 */
+	uint32_t flow_count;
+
+	/* Maximum number of schedule queues to be supported
+	 * Application configures the maximum number of schedule queues to be
+	 * supported by the implementation.
+	 * @see odp_queue_capability_t
+	 */
+	uint32_t queue_count;
+
+	/* Maximum number of events required to be stored simultaneously in
+	 * schedule queue. This number should be less than 'max_queue_size'
+	 * supported by the implementation.
+	 */
+	uint32_t queue_size;
+} odp_schedule_config_t;
+
+typedef struct odp_schedule_capability_t {
+
+	/* Maximum supported flows per queue
+	 * Specifies the maximum number of flows per queue supported by the
+	 * implementation.
+	 * A value of 0 indicates flow aware mode is not supported.
+	 */
+	uint32_t max_flow_count;
+
+	/* Maximum supported queues
+	 * Specifies the maximum number of queues supported by the
+	 * implementation.
+	 */
+	uint32_t max_queue_count;
+
+	/* Maximum number of events a schedule queue can store simultaneoulsy.
+	 * A value of 0 indicates the implementation does not restrict the
+	 * queue size
+	 */
+	uint32_t max_queue_size;
+} odp_schedule_capability_t;
+
+/**
+ * Start scheduler operation
+ *
+ * Activate scheduler module to schedule packets across different schedule
+ * queues. The scheduler module should be started before creating any odp
+ * queues. The scheduler module can be stopped usinig odp_schedule_stop().
+ *
+ * The initialization sequeunce should be,
+ * odp_schedule_capability()
+ * odp_schedule_config_init()
+ * odp_schedule_config()
+ * odp_schedule_start()
+ * odp_schedule()
```

Comment:
As we discussed, need to consider current applications that are happy with the current scheduler behavior. They shouldn't need to be changed to say "I don't need flow-aware scheduling".  We can either add a hint in the params for `odp_init_global()` that the application will use flow-aware mode or else say that the scheduler will operate by default in flow-unaware mode unless `odp_schedule_config()/odp_schedule_start()` is called.

Reply:
I believe it is easy to update in odp_init_global() saying that the default configuration of scheduler is flow-unware mode unless explicitly configured using odp_schedule_config()